src/test/java/com/github/dockerjava/cmd/PushImageCmdIT.java

```diff
@@ -50,30 +47,26 @@ public void pushLatest() throws Exception {
         String imageId = dockerRule.getClient().commitCmd(container.getId()).withRepository(username + "/busybox").exec();
 
         // we have to block until image is pushed
-        dockerRule.getClient().pushImageCmd(username + "/busybox").exec(new PushImageResultCallback()).awaitSuccess();
+        dockerRule.getClient().pushImageCmd(username + "/busybox").exec(new PushImageResultCallback()).awaitCompletion();
 
         LOG.info("Removing image: {}", imageId);
         dockerRule.getClient().removeImageCmd(imageId).exec();
 
-        dockerRule.getClient().pullImageCmd(username + "/busybox").exec(new PullImageResultCallback()).awaitSuccess();
+        dockerRule.getClient().pullImageCmd(username + "/busybox").exec(new PullImageResultCallback()).awaitCompletion();
     }
 
     @Test
     public void pushNonExistentImage() throws Exception {
-
-        if (getVersion(dockerRule.getClient())
-                .isGreaterOrEqual(RemoteApiVersion.VERSION_1_26)) {
-         // no errors??
-        } else if (getVersion(dockerRule.getClient())
-                .isGreaterOrEqual(RemoteApiVersion.VERSION_1_24)) {
-            exception.expect(DockerClientException.class);
-        } else {
+        //swarms throws a different error here
+        if (TestUtils.isSwarm(dockerRule.getClient())) {
```

Comment:
it not related to swarm. Run locally 2 DINDs with different version and run test against - you will see different behaviour

Reply:
The Problem was that sometimes awaitSuccess and sometimes awaitComplete was used. And they behaves inconsistent while handling the error. 
Therefore i propose that awaitComplete does the correct error handling and we deprecate awaitSuccess.

Reply:
i not sure at all why we parse messages because http response code should indicate success or failure...  But it could be changed in future, now i want end swarm & tests scope of changes and make major release.

Reply:
Not really since docker send always an success status code and will later then indicate the error status in the body. That is not so unknown pattern for long running asynchronous operations, since a http request will otherwise be timeout by clients or intermediate proxies.